instance/models/mixins/openedx_database.py

```diff
@@ -75,23 +75,17 @@ def mongo_database_names(self):
         """
         return [self.mongo_database_name, self.forum_database_name]
 
-    def provision_mysql(self):
+    def set_field_defaults(self):
         """
-        Set mysql credentials and provision the database.
+        Set default values for mysql and mongo credentials.
         """
-        if not self.mysql_provisioned:
-            self.mysql_user = get_random_string(length=16, allowed_chars=string.ascii_lowercase)
-            self.mysql_pass = get_random_string(length=32)
-        return super().provision_mysql()
-
-    def provision_mongo(self):
-        """
-        Set mongo credentials and provision the database.
-        """
-        if not self.mongo_provisioned:
-            self.mongo_user = get_random_string(length=16, allowed_chars=string.ascii_lowercase)
-            self.mongo_pass = get_random_string(length=32)
-        return super().provision_mongo()
+        if not self.use_ephemeral_databases:
+            if not self.mysql_provisioned:
```

Comment:
Since this is now a `set_field_defaults` method, I think this `if` should be `if not self.mysql_user:` (instead of checking `if not self.mysql_provisioned`). Otherwise multiple calls to `set_field_defaults()` would change the values of `mysql_user` and `mysql_pass` every time (until the DB has been provisioned).